frontend/src/features/Dashboard/components/DashboardForm/Reportings/Layer.tsx

```diff
@@ -0,0 +1,187 @@
+  )
+
+  const isPinned = selectedReportings?.includes(reporting)
+
+  const { subThemes, themes } = useGetControlPlans()
+
+  const ref = useRef<HTMLDivElement>(null)
+  const { newWindowContainerRef } = useNewWindow()
+  useClickOutsideEffect(
+    ref,
+    () => {
+      dispatch(dashboardActions.setSelectedReporting(undefined))
+    },
+    newWindowContainerRef.current
+  )
+
+  const handleSelect = e => {
+    e.stopPropagation()
+
+    if (isPinned) {
+      dispatch(dashboardActions.removeReporting(reporting))
+      dispatch(dashboardActions.setSelectedReporting(undefined))
+    } else {
+      dispatch(dashboardActions.addReporting(reporting))
+
+      focus(e)
+    }
+  }
+
+  const focus = e => {
+    e.stopPropagation()
+    dispatch(dashboardActions.setSelectedReporting(reporting))
+
+    const feature = new GeoJSON({
+      featureProjection: OPENLAYERS_PROJECTION
+    }).readFeature(reporting.geom)
+
+    const extent = feature?.getGeometry()?.getExtent()
+    if (extent) {
+      dispatch(setFitToExtent(extent))
+    }
+  }
+
+  const remove = e => {
+    e.stopPropagation()
+
+    dispatch(dashboardActions.removeReporting(reporting))
+  }
+
+  return (
+    <StyledReporting ref={ref} $isSelected={isSelected} onClick={e => focus(e)}>
+      <Header>
+        <Icon.Report
+          color={
+            reporting.reportType === ReportingTypeEnum.INFRACTION_SUSPICION
+              ? THEME.color.maximumRed
+              : THEME.color.blueGray
+          }
+        />
+        <Name>
+          Signalement {reporting.reportingId} -{' '}
+          {reporting.reportingSources.map(source => source.displayedSource).join(', ')}
+        </Name>
+
+        {isSelected ? (
+          <IconButton
+            accent={Accent.TERTIARY}
+            color={THEME.color.charcoal}
+            Icon={Icon.Close}
+            onClick={remove}
+            title="Supprimer le signalement"
+          />
+        ) : (
+          <IconButton
+            accent={Accent.TERTIARY}
+            color={isPinned ? THEME.color.blueGray : THEME.color.charcoal}
+            Icon={isPinned ? Icon.PinFilled : Icon.Pin}
+            onClick={handleSelect}
+            title="Sélectionner le signalement"
+          />
+        )}
+      </Header>
+      {reporting.createdAt && <Date>{getLocalizedDayjs(reporting.createdAt).format('DD MMM YYYY à HH:mm')} (UTC)</Date>}
+      <Content>
+        {reporting.themeId && (
+          <Theme>
+            {themes[reporting.themeId]?.theme} /{' '}
+            {reporting.subThemeIds.map(subThemeid => subThemes[subThemeid]?.subTheme).join(', ')} -{' '}
+          </Theme>
+        )}
+        <Description>{reporting.description}</Description>
+      </Content>
+    </StyledReporting>
+  )
+}
+
+const StyledReporting = styled.div<{ $isSelected: boolean }>`
+  background-color: ${p => p.theme.color.white};
+  display: flex;
+  flex-direction: column;
+
+  padding: 16px 24px;
+
+  :hover {
```

Comment:
beaucoup de saut de ligne dans ce style ;-) 
Et on met normalement une esperluette avant les `:focus` ou `:hover` => `&:focus`

Reply:
Si je me trompe pas l'esperluette c'est pour dire que le :hover s'applique uniquement au composant et pas ces enfants. ici ya que la div alors on s'en fiche non ? Je peux la mettre ça me gène pas ^^

Reply:
Et pour les sauts de ligne c'est une vieille habitude pour séparer les propriétés "par utilité". Généralement je groupe le positionnement (display, alignement), puis les margins / padding, puis les couleurs/border, puis les selectors css (hover, disabled...) puis le css in js (props). Mais je change pour pas mettre trop de ligne 🏊

Reply:
dans mon ancien projet on mettait juste des 🏊🏊🏊🏊🏊 en commentaire de PR ^^